src/BlazorSignalR/Internal/BlazorHttpMessageHandler.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Based on:
+// ASP.NET Core (https://github.com/aspnet/AspNetCore)
+// https://github.com/aspnet/AspNetCore/blob/v3.0.0-preview5-19227-01/src/Components/Blazor/Blazor/src/Http/WebAssemblyHttpMessageHandler.cs
+// https://github.com/aspnet/AspNetCore/blob/v3.0.0-preview5-19227-01/src/Components/Blazor/Blazor/src/Http/FetchCredentialsOption.cs
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. 
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace BlazorSignalR.Internal
+{
+    /// <summary>
+    /// A browser-compatible implementation of <see cref="HttpMessageHandler"/>
+    /// </summary>
+    internal class BlazorHttpMessageHandler : HttpMessageHandler
```

Comment:
As mentioned already, I see little purpose for having this this heavy weight custom ```HttpMessageHandler``` just for a one-off negotiation request.

Reply:
The `BlazorServerSentEventsTransport` internally also uses an http-client. Replacing the current solution needs us to rewrite the negotiation in js together with the parts the `BlazorServerSentEventsTransport` does with it. I don't think this will be less costly for at least the development and maintenance costs. For runtime statistics we had to test this as I cannot estimate the performance characteristics of the two solutions. The current solution also gives as the benefit if we ever wanted to support some kind of managed (well C# implemented) transport to have a solution for HTTP-transmission out of the box.

Reply:
Hmmm, I'm quite torn about this.

One side of me thinks when on server side blazor, we should never use c# implemented transports because this risks the request being sent from server (API might change etc) is too high, and we can tune the performance more of the custom JS implementations, therefore removing the need for this large setup. The negotiation code should be quite short (we can look at the typescript client code), and I imagine the usage of it in the server sent event transport is quite minimal/easy to convert.

Though I also think that this could be useful in the future, for when we need to bodge some more functionality, or maybe someone wants to add a custom transport.

Reply:
I suggest to retain the current solution. When we ever need to, due to perfromance problems or other issues, we can investigate with replacing the current solution (partially) with a js-implemented one. This also does not bloat this PR with too many changes that are not actually necessary to implement the desired set of features.

Reply:
Via that argument, I would argue ```BlazorHttpMessageHandler``` is more bloat and unneeded for the actual features we need. It is going to become a large burden to maintain, as we will have to track the Blazor repo to see if any bugs etc are detected with the more or less copied implementation?

You can either remove it, or I can if you prefer.